src-tauri/src/mod_management/mod.rs

```diff
@@ -383,3 +385,46 @@ pub async fn fc_download_mod_and_install(
 
     Ok(())
 }
+
+/// Deletes a given Northstar mod folder
+fn delete_mod_folder(ns_mod_directory: String) -> Result<(), String> {
+    // TODO maybe remove again
+    let ns_mod_dir_path = std::path::Path::new(&ns_mod_directory);
+
+    // Safety check: Check whether `mod.json` exists and exit early if not
+    // If it does not exist, we might not be dealing with a Northstar mod
+    let mod_json_path = ns_mod_dir_path.join("mod.json");
+    if !mod_json_path.exists() {
+        // If it doesn't exist, return an error
+        return Err(format!("mod.json does not exist in {}", ns_mod_directory));
+    }
+
+    match std::fs::remove_dir_all(&ns_mod_directory) {
+        Ok(()) => Ok(()),
+        Err(err) => Err(format!("Failed deleting mod: {err}")),
+    }
+}
+
+/// Deletes a Northstar mod based on its name
+pub fn delete_northstar_mod(game_install: GameInstall, nsmod_name: String) -> Result<(), String> {
+    // Prevent deleting core mod
```

Comment:
Additionally to throwing an error when trying to remove a core mod, the "delete" button should not appear on core mod tiles in UI.

Reply:
Hmm, for that we need to signal core mods from backend to front-end. I don't think we have something for that yet... :/

Reply:
I thought we had a list of core mods somewhere? Only in backend maybe?

Reply:
Moved to #120